### DISCUSSION TITLE
Is it possible to filter a PrimaryKeyRelatedField?

### DISCUSSION THREAD

**Discussion by kevinrenskers** [encode/django-rest-framework#8195] [Question & Answer]
Is is possible to filter the array of results of a PrimaryKeyRelatedField? For example, take this serializer: 

``` python
class QuestSerializer(DmNotesModelSerializer):
    characters = CharacterPrimaryKeyRelatedField(many=True)

    class Meta:
        model = Quest
        fields = "__all__"
```

Currently it always includes all the characters per quest, but how can I add a filter to that? In my case characters can be hidden, but why if you use a soft-delete method and have an `is_deleted` property on the character model, and you want to exclude such characters from the quest(s)?

For more background on this question you could check out [this StackOverflow](https://stackoverflow.com/questions/69417838/django-rest-frameworking-modify-primarykeyrelatedfield-queryset-for-get-request) question that I've created which has a lot more details.

**Comment by Hemanthkumar2112** [comment]
use [https://www.django-rest-framework.org/api-guide/fields/#serializermethodfield](Serializermethodfield) or else we can override PrimaryKeyRelatedField class with get_queryset()
